src/librbd/mirror/EnableRequest.cc

```diff
@@ -68,27 +68,33 @@ void EnableRequest<I>::handle_get_mirror_image(int r) {
     r = cls_client::mirror_image_get_finish(&iter, &m_mirror_image);
   }
 
-  if (r == 0 && m_mirror_image.state == cls::rbd::MIRROR_IMAGE_STATE_CREATING &&
-      !m_non_primary_global_image_id.empty()) {
-    // special case where rbd-mirror injects a disabled record to record the
-    // local image id prior to creating ther image
```

Comment:
As this PR is supposed to just strengthen the conditions without changing the core logic, I'd preserve this comment.

Reply:
I will remove the typo in the comment s/ther/the

However,  I find the comment confusing. The rbd-mirror doesn't inject a disabled record, the mirror image record is in CREATING state. Does the following rephrasing of the comment make sense to you,
" special case where rbd-mirror injects a mirror image record to preserve the local image id prior to creating the image" ?

Reply:
I might have suggested "special case where rbd-mirror injects a quasi-enabled mirror image entry to record the local image id prior to creating the image", but yours is fine too.